Automattic/jetpack
# Jetpack: gracefully fail when user data fetch failed

## Proposed changes:
When Jetpack failed to fetch user data from WPCOM, Jetpack Dashboard and My Jetpack display invalid output.
This PR will make it fail gracefully without JS errors, and display a proper error message.

### Current output:
<img width="528" alt="Screenshot 2024-09-01 at 20 30 07" src="https://github.com/user-attachments/assets/e6d2790e-7b35-47f9-a4d3-354ae66fe960">

<img width="359" alt="Screenshot 2024-09-01 at 20 34 12" src="https://github.com/user-attachments/assets/eecfafaa-9011-47b8-944e-1b92b9f4a26c">

### New output:
<img width="524" alt="Screenshot 2024-09-01 at 20 56 54" src="https://github.com/user-attachments/assets/4574061e-c7f0-4631-b842-937074822b02">
<img width="301" alt="Screenshot 2024-09-01 at 20 57 44" src="https://github.com/user-attachments/assets/7d6eda12-7b6f-4465-93b4-af3db3988f1b">


### Other information:


## Does this pull request change what data or activity we track or use?
No.

## Testing instructions:
1. Connect Jetpack site, switch it to a non-existing sandbox URL.
2. Load Jetpack Dashboard, confirm that the "Account Connection" card properly displays avatar, WPCOM username, and email.
3. Load My Jetpack, confirm that the "Connection" card displays the WPCOM username: "Connected as [username] (manage)".
4. Delete the transient with user data stored: `wp transient delete jetpack_connected_user_data_1` (where `1` is your WordPress user ID).
5. Load Jetpack Dashboard, confirm that the "Account Connection" card matches the "New output" screenshot.
6. Load My Jetpack, confirm that user connection data matches the "New output" screenshot.
7. Connect another WP user account as secondary user, repeat steps 2-6.
